### ISSUE TITLE
AutoDj runs for a few songs then stops.

### ISSUE THREAD

**Issue by mikeopensauce** [AzuraCast/AzuraCast#4133]
**Using Docker installation method**
Yes

**AzuraCast version**
v0.12.4 stable

**Host Operating System**
Ubuntu 20.04.2 LTS

**Describe the bug**
AutoDJ service is running (according to GUI and reports " LiquidSoap is currently shuffling from 91 uploaded songs and 3 playlists.". However the player shows now playing "AzuraCast is Live!". I have tried:
- Restarting liquidsoap
- Restarting icecast
- Restarting the server
- Updating the server OS
- Updating Azuracast to latest version
- Backing up station, removing Azuracast, reinstalling and restoring station

Interestingly when I restart Icecast the AuotDj starts again and plays a few tracks then stops again reverting to playing "AzuraCast is Live!"

**To Reproduce**
1. Start Azuracast broadcasting

**Expected behavior**
Auto Dj plays songs and other audio files from playlists

**Relevant Logs**
Snippet from Liquidsoap log showing point where playing was working then stopped.

2021/05/06 16:51:24 [playlist_interviews:3] Prepared "/var/azuracast/stations/rosetta_radio/media/interviews/mc-martinwhite.mp3" (RID 3).
2021/05/06 16:51:24 [playlist_musiclicencefree:3] Prepared "/var/azuracast/stations/rosetta_radio/media/musiclicencefree/10_-_track_10.ogg" (RID 4).
2021/05/06 16:51:24 [autodj_fallback:3] Switch to standard_playlists with forgetful transition.
2021/05/06 16:51:24 [standard_playlists:3] Switch to sequence_38751.
2021/05/06 16:51:24 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/rosetta_radio/media/interviews/mdb-alemaryspackage.mp3".
2021/05/06 16:51:24 [cross_38790:3] Analysis: -infdB / -20.264679dB (2.96s / 2.96s)
2021/05/06 16:51:24 [crossfade:3] new >= old + margin, old <= medium and new <= high.
2021/05/06 16:51:24 [crossfade:3] Do not fade if it's already very low.
2021/05/06 16:51:24 [crossfade:3] Transition: crossed, no fade.
2021/05/06 16:51:24 [lang:3] AzuraCast Feedback Response: OK
2021/05/06 16:52:55 [standard_playlists:3] Switch to sequence_38753 with forgetful transition.
2021/05/06 16:52:55 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/rosetta_radio/media/musiclicencefree/10_-_track_10_-_cd1.ogg".
2021/05/06 16:52:55 [lang:3] AzuraCast Raw Response: 
2021/05/06 16:52:55 [cross_38790:3] Analysis: -30.844299dB / -19.560027dB (2.99s / 3.01s)
2021/05/06 16:52:55 [crossfade:3] new >= old + margin, old >= medium and new <= high.
2021/05/06 16:52:55 [crossfade:3] New source is significantly louder than old one.
2021/05/06 16:52:55 [crossfade:3] Transition: crossed, fade-out.
2021/05/06 16:52:55 [lang:3] AzuraCast Feedback Response: OK
2021/05/06 16:56:56 [cross_38790:3] Analysis: -nandB / -nandB (2.98s / 0.00s)
2021/05/06 16:56:56 [crossfade:3] Old <= medium, new <= medium and |old-new| <= margin.
2021/05/06 16:56:56 [crossfade:3] Old and new source are not too loud and close.
2021/05/06 16:56:56 [crossfade:3] Transition: crossed, fade-in, fade-out.
2021/05/06 16:56:56 [lang:3] AzuraCast Raw Response: 
2021/05/06 16:56:56 [lang:3] AzuraCast Feedback Response: OK
2021/05/06 16:56:56 [lang:3] AzuraCast Raw Response: 
2021/05/06 16:56:58 [lang:3] AzuraCast Raw Response: 
2021/05/06 16:56:59 [lang:3] AzuraCast Raw Response: 
2021/05/06 16:56:59 [error_jingle:3] Prepared "/usr/local/share/icecast/web/error.mp3" (RID 0).
2021/05/06 16:56:59 [safe_fallback:3] Switch to error_jingle with forgetful transition.
2021/05/06 16:56:59 [ffmpeg:3] [mp3float @ 0x7f9ec07a11c0] Could not update timestamps for skipped samples.
2021/05/06 16:57:01 [lang:3] AzuraCast Raw Response: 

**Device(s):**

- Device: Desktop
- OS: Ubuntu 20.04
- Browser: Firefox
- Version: 88

**Additional context**
Icecast error log almost empty . Other logs available on request

**Comment by SlvrEagle23** [comment]
@mikeopensauce The "AzuraCast Raw Response: " blank lines are what's concerning in this case.

From `System Administration`, then `Log Viewer`, look at the AzuraCast application log (the top entry on that page) and copy/paste any errors that appear there; particularly errors to do with the AutoDJ.

It's possible those errors may explain what's going wrong in a way that you can resolve within your own configuration, but if not we can take a look at them and see what's going on.

**Comment by mikeopensauce** [comment]
My apologies, I missed that second log location.
That log is full of errors. Sample below.

[2021-05-06T16:27:56.411683+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:56.412164+00:00] AzuraCast.ERROR: Too many attempts to get next song; giving up. {"station":{"id":1,"name":"Rosetta Radio"}} []
[2021-05-06T16:27:58.505852+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.510478+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.514842+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.515342+00:00] AzuraCast.ERROR: Too many attempts to get next song; giving up. {"station":{"id":1,"name":"Rosetta Radio"}} []
[2021-05-06T16:27:58.591377+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.596006+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.600315+00:00] AzuraCast.ERROR: No valid playlists detected. Skipping AutoDJ calculations. [] {"station":{"id":1,"name":"Rosetta Radio"}}
[2021-05-06T16:27:58.600772+00:00] AzuraCast.ERROR: Too many attempts to get next song; giving up. {"station":{"id":1,"name":"Rosetta Radio"}} []

**Comment by mikeopensauce** [comment]
Thanks for your help
.
The songs were uploaded in the GUI and the playlists were created in there as well. See screenshots attached.
If that is the case I wonder if just deleting the playlists and recreating them may work as it could just be some sort of corruption somewhere. It was running fine for a couple of years but just died recently.

![songsscreenshot](https://user-images.githubusercontent.com/7012099/117334876-6e110700-ae92-11eb-829c-224504eec5b6.png)

![playlistscreenshot](https://user-images.githubusercontent.com/7012099/117334892-710bf780-ae92-11eb-9e46-36bf9c1a167b.png)

**Comment by SlvrEagle23** [comment]
@mikeopensauce And none of the playlists are scheduled?

The error message is specifically saying that there are no _valid_ playlists detected; it could be that they're using advanced settings, or otherwise are configured in such a way that the system doesn't handle them normally.

**Comment by mikeopensauce** [comment]
Thank you. I'll try deleting them and rebuilding them and see how I get on. Thanks for narrowing down my search.

Mike

**Comment by mikeopensauce** [comment]
As an update, I deleted the playlists and recreated them but with the same effect. Taking on your advice I stripped them down and I believe I have resolved it. I only wanted on track to play from each playlist before moving on to the next playlist i.e. Play 1 song then play 1 interview then play 1 song , etc.
To achieve this I checked the box that says "Only play one track at scheduled time." When I removed that selection it solved the problem.
Thank you for pointing me in the right direction and keep up the good work!
Mike
